pkg/prometheus/statefulset.go

```diff
@@ -437,22 +410,21 @@ func makeStatefulSetSpec(
 	if p.Spec.LogLevel != "" && p.Spec.LogLevel != "info" {
 		promArgs = append(promArgs, monitoringv1.Argument{Name: "log.level", Value: p.Spec.LogLevel})
 	}
-	if version.GTE(semver.MustParse("2.6.0")) {
-		if p.Spec.LogFormat != "" && p.Spec.LogFormat != "logfmt" {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "log.format", Value: p.Spec.LogFormat})
-		}
+
+	if p.Spec.LogFormat != "" && p.Spec.LogFormat != "logfmt" {
+		promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "log.format", Value: p.Spec.LogFormat})
 	}
 
-	if version.GTE(semver.MustParse("2.11.0")) && p.Spec.WALCompression != nil {
+	if p.Spec.WALCompression != nil {
+		arg := monitoringv1.Argument{Name: "no-storage.tsdb.wal-compression"}
 		if *p.Spec.WALCompression {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.wal-compression"})
-		} else {
-			promArgs = append(promArgs, monitoringv1.Argument{Name: "no-storage.tsdb.wal-compression"})
+			arg.Name = "storage.tsdb.wal-compression"
 		}
+		promArgs = cg.WithMinimumVersion("2.11.0").AppendCommandlineArgument(promArgs, arg)
 	}
 
-	if version.GTE(semver.MustParse("2.8.0")) && p.Spec.AllowOverlappingBlocks {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "storage.tsdb.allow-overlapping-blocks"})
+	if p.Spec.AllowOverlappingBlocks {
+		promArgs = cg.WithMinimumVersion("2.11.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "storage.tsdb.allow-overlapping-blocks"})
```

Comment:
(nit) since 2.39.0, this flag has no effect and is deprecated.

Reply:
I've added `.WithMaximumVersion("2.39.0")`